### DISCUSSION TITLE
Show loader over suspense boundary

### DISCUSSION THREAD

**Discussion by jnovak-SM2Dev** [vercel/next.js#62666] [Help]
### Summary

What is the proper way to show a loader over a suspense boundary? I know you can set a `fallback` for `Suspense`, but that's only helpful on the first page load. If i'm doing pagination for example, how can I have it so the fallback works for the first load, but then show a loader over the previous results while the new results load (via server action)?

**Comment by ancashoria** [comment]
Hey,
I searched for a solution and I have found [this in the react docs](https://react.dev/reference/react/Suspense#resetting-suspense-boundaries-on-navigation) .

For me it worked like this:
```
<Suspense key={JSON.stringify(serverSearchParams)} fallback={...}>...</Suspense>
```

**Comment by jnovak-SM2Dev** [comment]
@ancashoria That is what I already have and it just replaces the content in the suspense with the fallback. What i'd like to do is show a loader over the previous content until the new content is ready. Basically if you have a table with data in it, i'd like when you click on next page for the current page's data to stay, but show a loader on top until the new data is ready and then switch it.
